Add `&nomutepreview` option and apply `&volume` to previews

comex opened the pull request:
When streaming video, it was already possible to monitor input audio by showing video controls for the local video preview and pressing the unmute button.  This PR adds two features to enhance this functionality:

- Add an option, `&nomutepreview`, to automatically unmute the preview.

  If passed in sender mode, the preview still starts muted on the device selection screen, but gets unmuted when you press "start".  This is to avoid undesired results if the wrong audio device is preselected.

  The option can also be passed in director mode in order to monitor the director's own stream.  In this case it takes effect as soon as the director enables their microphone/video, since there's no separate device selection screen.

- Apply the `&volume` setting to local video previews (again, in both sender and director mode).

  By itself, this does not unmute the preview, but determines what volume you get if you press the unmute button.  This can also be combined with `&nomutepreview`.

Documentation is included, but I wasn't able to test the formatting (I don't know how to replicate the Markdown-to-HTML process used for the docs site).

Features _not_ included:

- Monitoring a subset of input audio devices

- Sending monitor output to a non-default audio device

Still, this is good enough for me.  My use case involves setting an Elgato HDMI capture device as the video/audio input in order to stream gameplay (directly through vdo.ninja, no OBS).  I need to monitor audio in order to hear the game.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/steveseguin/vdo.ninja?pullRequest=1198) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/steveseguin/vdo.ninja?pullRequest=1198) before we can accept your contribution.<br/><hr/>**comex** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/steveseguin/vdo.ninja?pullRequest=1198) it.</sub>

comex committed:
Add `&nomutepreview` option and apply `&volume` to previews

When streaming video, it was already possible to monitor input audio by
showing video controls for the local video preview and pressing the
unmute button.  This PR adds two features to enhance this functionality:

- Add an option, `&nomutepreview`, to automatically unmute the preview.

  If passed in sender mode, the preview still starts muted on the device
  selection screen, but gets unmuted when you press "start".  This is to
  avoid undesired results if the wrong audio device is preselected.

  The option can also be passed in director mode in order to monitor the
  director's own stream.  In this case it takes effect as soon as the
  director enables their microphone/video, since there's no separate
  device selection screen.

- Apply the `&volume` setting to local video previews (again, in both
  sender and director mode).

  By itself, this does not unmute the preview, but determines what
  volume you get if you press the unmute button.  This can also be
  combined with `&nomutepreview`.

Documentation is included, but I wasn't able to test the formatting (I
don't know how to replicate the Markdown-to-HTML process used for the
docs site).

Features _not_ included:

- Monitoring a subset of input audio devices

- Sending monitor output to a non-default audio device

Still, this is good enough for me.  My use case involves setting an
Elgato HDMI capture device as the video/audio input in order to stream
gameplay (directly through vdo.ninja, no OBS).  I need to monitor audio
in order to hear the game.

steveseguin commented:
Thank you for the contribution.  I can't blindly accept code changes of course, I wish I could, so I'll need to find time to review things closely. If all goes well, I'll either merge these changes or push my own version of the features to github and to production.
